pandas/core/indexing.py

```diff
@@ -1814,7 +1814,8 @@ def check_bool_indexer(ax, key):
         result = result.reindex(ax)
         mask = isnull(result._values)
         if mask.any():
-            raise IndexingError('Unalignable boolean Series key provided')
+            raise IndexingError('Unalignable labels in boolean Series index '
+                                '{}'.format(key.index))
```

Comment:
I am not sure if this will give a nice result (adding the key's index in the error message, the rewording is certainly an improvement!)

Can you show an example and its resulting error message?